### PR TITLE
Migration from AWSMachineClass to MachineClass

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:
This code adds the migration logic to move from AWSMachineClass to MachineClass.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy user
Migration of `AWSMachineClass` to `MachineClass` CRs is implicity. Don't use old AWSMachineClass CRs post-migration.
```

### CODE REVIEW

**Comment by prashanth26** [comment]
PR is similar to - https://github.com/gardener/machine-controller-manager-provider-gcp/pull/5. 
/invite @amshuman-kr @hardikdr

**Review by AxiomSamarth — APPROVED** [review]

File: pkg/aws/core.go [hunk 1]
```diff
@@ -364,3 +374,19 @@ func (d *Driver) GetVolumeIDs(ctx context.Context, req *driver.GetVolumeIDsReque
 	}
 	return resp, nil
 }
+
+// GenerateMachineClassForMigration converts providerSpecificMachineClass to (generic) MachineClass
+func (d *Driver) GenerateMachineClassForMigration(ctx context.Context, req *driver.GenerateMachineClassForMigrationRequest) (*driver.GenerateMachineClassForMigrationResponse, error) {
+	klog.V(1).Infof("Migrate request has been recieved for %v", req.MachineClass.Name)
+	defer klog.V(1).Infof("Migrate request has been processed for %v", req.MachineClass.Name)
+
+	gcpMachineClass := req.ProviderSpecificMachineClass.(*v1alpha1.AWSMachineClass)
```
Comment (hardikdr) [inline pkg/aws/core.go:383]: s/gcpMachineClass/awsMachineClass/g. ;)

File: pkg/aws/core_util_test.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+						VolumeType:          aws.String("gp2"),
+					},
+				},
+			}
+
+			Expect(disksGenerated).To(Equal(expectedDisks))
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should convert zero blockDevices successfully", func() {
```
Comment (hardikdr) [inline pkg/aws/core_util_test.go:198]: Shouldn't it rather fail, if there are no disks at all provided?

**Review by hardikdr — COMMENTED** [review]
Looks good overall, just minor comments.
Thanks for the PR.

File: pkg/aws/apis/validation/validation.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func ValidateAWSProviderSpec(spec *awsapi.AWSProviderSpec, secrets *corev1.Secre
 
 	allErrs = append(allErrs, validateBlockDevices(spec.BlockDevices)...)
 	allErrs = append(allErrs, validateNetworkInterfaces(spec.NetworkInterfaces)...)
-	allErrs = append(allErrs, validateSecrets(secrets)...)
+	allErrs = append(allErrs, ValidateSecret(secret)...)
```
[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/aws/apis/validation/validation.go:57]: Yes this was being called from on of the other packages. So - https://github.com/gardener/machine-controller-manager-provider-aws/pull/15/files#diff-2e818d219fe1bb06d7f672104bebaa13de632d1239a7fb0873e3e13b55adfadcR187.

[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/aws/apis/validation/validation.go:57]: Yes this was being called from on of the other packages. So - https://github.com/gardener/machine-controller-manager-provider-aws/pull/15/files#diff-2e818d219fe1bb06d7f672104bebaa13de632d1239a7fb0873e3e13b55adfadcR187.

**Review by prashanth26 — COMMENTED** [review]
Made other suggestions here - https://github.com/gardener/machine-controller-manager-provider-aws/pull/15/commits/b42f4ed639a7ec300f3b906521f42880d15de796

**Review by hardikdr — APPROVED** [review]

**Pull request merged**